### PR TITLE
Validate reactions

### PR DESCRIPTION
**Pull request by Vertonowsky**
### Reactions
Reaction's type must be now registered in `ReactionType` enum. That will be the proper way to validate already exisitng reactions. 

### CreateRoom and JoinRoom reactions
Implemented complete process of **validation** and insertion to **database**.

### Exceptions
Added reaction's specific exceptions. All exceptions will be now storaged in the same package as they are used. 

Also there was a update on `User` and `Room` model which includes equals() and hashCode() functions.

### CODE REVIEW

File: src/main/java/com/github/splendor_mobile_game/websocket/handlers/reactions/CreateRoom.java [hunk 1]
```diff
@@ -85,42 +90,56 @@ public void react(ReceivedMessage parsedMessage, Messenger messenger, Database d
 
             JsonObject response = new JsonObject();
             response.addProperty("messageContextId", parsedMessage.getMessageContextId());
-            response.addProperty("type", "CreateRoomResponse");
-            response.addProperty("result", "OK");
+            response.addProperty("type", ResponseType.CREATE_ROOM_RESPONSE.toString());
```
Comment (kestivvi) [inline src/main/java/com/github/splendor_mobile_game/websocket/handlers/reactions/CreateRoom.java:96]: So this `ResponseType.CREATE_ROOM_RESPONSE.toString()` will result in `CREATE_ROOM_RESPONSE` string?

File: src/main/java/com/github/splendor_mobile_game/websocket/handlers/reactions/CreateRoom.java [hunk 2]
```diff
@@ -64,16 +67,18 @@ public void react(ReceivedMessage parsedMessage, Messenger messenger, Database d
...
 
         try {
-            validateData(receivedMessage);
+            validateData(receivedMessage, database);
 
-            User user = new User(receivedMessage.userDTO.id, receivedMessage.userDTO.name, this.connectionHashCode);
+            User user = new User(receivedMessage.userDTO.uuid, receivedMessage.userDTO.name, this.connectionHashCode);
             Room room = new Room(UUID.randomUUID(), receivedMessage.roomDTO.name, receivedMessage.roomDTO.password, user);
 
+            System.out.println("UUID pokoju: " + room.getUuid());
```
Comment (kestivvi) [inline src/main/java/com/github/splendor_mobile_game/websocket/handlers/reactions/CreateRoom.java]: I propose using static methods of `Log` class. You could write there something as `Log.DEBUG("UUID pokoju: " + room.getUuid()`, then this print would be blue on the terminal and simultaneously it'll be written to the log file. There are multiple levels of logs: `Log.TRACE`, `Log.DEBUG`, `Log.WARNING` and `Log.ERROR`.

File: src/main/java/com/github/splendor_mobile_game/websocket/handlers/reactions/CreateRoom.java (hunk 1 above)
Comment (Vertonowsky) [reply src/main/java/com/github/splendor_mobile_game/websocket/handlers/reactions/CreateRoom.java:96]: That's right. There is an option to write custom .toString() function which will return `CreateRoomResponse` String but in my opinion it's better to transfer `type` excatly as ENUM looks like. Same thing on ClientSide. It increases readability.

Comment (kestivvi) [reply src/main/java/com/github/splendor_mobile_game/websocket/handlers/reactions/CreateRoom.java:96]: Currently client has to write type of request the same as the name of the class extending `Reaction`. So if we want change convention into upper case snake case, then i have to change framework a little to make an option to specify different name than the class name or... and that's ridiculous, we have to write class names extending reaction like CREATE_ROOM 🤣.

If you want, i can change the framework a little bit to make custom names possible. Actually i did something like this earlier, but i forgo it.

File: src/main/java/com/github/splendor_mobile_game/websocket/handlers/reactions/CreateRoom.java (hunk 2 above)
Comment (Vertonowsky) [reply src/main/java/com/github/splendor_mobile_game/websocket/handlers/reactions/CreateRoom.java]: I accept the proposal

File: src/main/java/com/github/splendor_mobile_game/websocket/handlers/reactions/CreateRoom.java (hunk 1 above)
Comment (Vertonowsky) [reply src/main/java/com/github/splendor_mobile_game/websocket/handlers/reactions/CreateRoom.java:96]: So right now when user is listening for response from the server, his code should contain a class named `CREATE_ROOM_RESPONSE`? Am i thinking right?

Comment (kestivvi) [reply src/main/java/com/github/splendor_mobile_game/websocket/handlers/reactions/CreateRoom.java:96]: No no no. Currently user can have classes whatever it wants to.

If user sends request to the server with type `CREATE_ROOM`, then server has to have reaction class named `CREATE_ROOM`.

**Pull request merged**